TGeant3/TGeant3.cxx

```diff
@@ -6797,10 +6800,11 @@ extern "C" void type_of_call rxgtrak(Int_t &mtrack, Int_t &ipart, Float_t *pmom,
          polar[0] = pol.X();
          polar[1] = pol.Y();
          polar[2] = pol.Z();
+         if (track->IsPrimary() || track->GetUniqueID() == 0) {
```

Comment:
The test for "track->IsPrimary()" is redundant, as a particle can be injected into MC with an existing mother (what happens in AliRoot), and then return IsPrimary() false, what was btw the problem with the current implementation.
The test for " track->GetUniqueID()" uses a user (in this case AliRoot) specific implementation, where the creator process is saved as TParticle uniqueID. This may however be not true in other stack implementation.
So I suggest to go back here to the original logic, which was based on the event processing loop that guaranties that all primaries for an event are created before all secondaries, and so each secondary track ID is >= than nofPrimaries.

Reply:
Ok. Also this I can do as soon as I have the changes necessary to address your first Fortran comment